libcloud/compute/drivers/gig_g8.py

```diff
@@ -0,0 +1,501 @@
+        return True
+
+    def start_node(self, node):
+        """
+        Start virtual machine
+        """
+        node.state = NodeState.STARTING
+        self._api_request("/machines/start", {"machineId": int(node.id)})
+        node.state = NodeState.RUNNING
+        return True
+
+    def ex_list_networks(self):
+        """
+        Return the list of networks.
+
+        :return: A list of network objects.
+        :rtype: ``list`` of :class:`G8Network`
+        """
+        networks = []
+        for network in self._api_request("/cloudspaces/list"):
+            if network["accountId"] == self._account_id:
+                networks.append(self._to_network(network))
+        return networks
+
+    def list_sizes(self):
+        """
+        Returns a list of node sizes as a cloud provider might have
+
+        """
+        location = self._location["locationCode"]
+
+        sizes = []
+        for size in self._api_request("/sizes/list", {"location": location}):
+            sizes.extend(self._to_size(size))
+        return sizes
+
+    def list_nodes(self, network=None):
+        """
+        List the nodes known to a particular driver;
+        There are two default nodes created at the beginning
+        """
+        if network:
+            networks = [network]
+        else:
+            networks = self.ex_list_networks()
+        nodes = []
+        for network in networks:
+            nodes_list = self._api_request("/machines/list",
+                                           params={"cloudspaceId": network.id})
+            for nodedata in nodes_list:
+                nodes.append(self._to_node(nodedata))
+        return nodes
+
+    def reboot_node(self, node):
+        """
+        Reboot node
+        returns True as if the reboot had been successful.
+        """
+        node.state = NodeState.REBOOTING
+        self._api_request("/machines/reboot", {"machineId": int(node.id)})
+        node.state = NodeState.RUNNING
+        return True
+
+    def destroy_node(self, node):
+        """
+        Destroy node
+        """
+        self._api_request("/machines/delete", {"machineId": int(node.id)})
+        return True
+
+    def list_images(self):
+        """
+        Returns a list of images as a cloud provider might have
+
+        @inherits: :class:`NodeDriver.list_images`
+        """
+        images = []
+        for image in self._api_request("/images/list",
+                                       {"accountId": self._account_id}):
+            images.append(self._to_image(image))
+        return images
+
+    def list_volumes(self):
+        volumes = []
+        for disk in self._api_request("/disks/list",
+                                      {"accountId": self._account_id}):
+            if disk["status"] not in ["ASSIGNED", "CREATED"]:
+                continue
+            volumes.append(self._to_volume(disk))
+        return volumes
+
+    def create_volume(self, size, name, description, disk_type="D"):
```

Comment:
``description`` and ``disky_type`` arguments are not part of the base API so please prefix them with ``ex_``.